term/src/terminalstate/image.rs

```diff
@@ -202,6 +221,11 @@ impl TerminalState {
             }
         }
 
+        // adjust cursor position if the drawn cells move beyond current cell
+        let x_padding_shift = (draw_width as usize + cell_padding_left as usize
+            > cell_pixel_width * width_in_cells) as i64;
+        let y_padding_shift = (draw_height as usize + cell_padding_top as usize
+            > cell_pixel_height * height_in_cells) as i64;
```

Comment:
I think these are candidates for `one_or_zero` as well:

```suggestion
        let x_padding_shift = one_or_zero(draw_width as usize + cell_padding_left as usize
            > cell_pixel_width * width_in_cells);
        let y_padding_shift = one_or_zero(draw_height as usize + cell_padding_top as usize
            > cell_pixel_height * height_in_cells);
```